Changes in module 'nf-core/multiqc'
--- modules/nf-core/multiqc/main.nf
+++ modules/nf-core/multiqc/main.nf
@@ -1,7 +1,7 @@
 process MULTIQC {
     tag "${meta.id}"
     label 'process_single'
-
+    tag "$meta.model"
     conda "${moduleDir}/environment.yml"
     container "${workflow.containerEngine in ['singularity', 'apptainer'] && !task.ext.singularity_pull_docker_container
         ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/c8/c8e346f4f6080eadf1253505e6ff09ef004454fc18e8d672006fd7b222cc412e/data'
@@ -28,7 +28,8 @@ process MULTIQC {
     def replace = replace_names ? "--replace-names ${replace_names}" : ''
     def samples = sample_names ? "--sample-names ${sample_names}" : ''
     """
-    multiqc \\
+    pip install --target "\$PWD/.multiqc_plugins" ${workflow.projectDir}
+    PYTHONPATH="\$PWD/.multiqc_plugins" multiqc \\
         --force \\
         ${args} \\
         ${config} \\
